src/DAL/migration/storedProcedure.sql

```diff
@@ -0,0 +1,75 @@
+CREATE OR REPLACE PROCEDURE insert_polygon_part(r "PolygonParts".insert_polygon_part_record)
+LANGUAGE plpgsql
+AS $$
+DECLARE
+    is_valid_result RECORD;
+    is_valid boolean;
+    reason text;
+BEGIN
+    -- check validity of the input polygon geometry
+    is_valid_result := ST_IsValidDetail(r.geom);
+
+    is_valid := is_valid_result.valid;
+    reason := is_valid_result.reason;
+
+    IF NOT is_valid THEN
+        RAISE EXCEPTION 'Invalid geometry: %', reason;
+    END IF;
+
+    -- check that input polygon extent is within the bbox of the srs (EPSG:4326)
+    is_valid := ST_Extent(r.geom)@Box2D(ST_GeomFromText('LINESTRING(-180 -90, 180 90)'));
+
+    IF NOT is_valid THEN
+        RAISE EXCEPTION 'Invalid geometry extent: %', ST_Extent(r.geom);
+    END IF;
+
+    -- create (if not exists) a temp table to hold results instersection result for the current input
+    CREATE TEMPORARY TABLE IF NOT EXISTS selected_parts ON COMMIT DROP AS (
+        SELECT ST_IsEmpty(geom_intersections) is_empty_geom, ST_NumGeometries(geom_intersections) > 1 is_multi, *
+        FROM (
+            SELECT "internalId", ST_Difference(geom, r.geom) geom_intersections
+            FROM "PolygonParts".parts 
+            WHERE ST_Intersects(geom, r.geom)
+        ) q
+    ) WITH NO DATA;
+
+    -- insert intersecting polygon parts into a temp table
+    INSERT INTO selected_parts
+	SELECT ST_IsEmpty(geom_intersections) is_empty_geom, ST_NumGeometries(geom_intersections) > 1 is_multi, *
+    FROM (
+        SELECT "internalId", ST_Difference(geom, r.geom) geom_intersections
+        FROM "PolygonParts".parts 
+        WHERE ST_Intersects(geom, r.geom)
+    ) q;
```

Comment:
This is used twice, maybe I don't understand why this is needed.

Reply:
for the first time it defines the structure of the table, hence `WITH NO DATA`
the second time it inserts the data
it was refactored because when the query was calculated within `create temporary table if not exists` it was skipped over from the second input polygon, because the table already existed